lighthouse-core/gather/gatherers/forms.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+
+/**
+ *  @param {HTMLFormElement} formElement
+ *  @return { { id: any; nodeName: any; name: any; placeholder: any; autocomplete: any; }[] }
+ */
+/* istanbul ignore next */
+function getChildrenInputs(formElement) {
+  /** @type {{ id: any; nodeName: any; name: any; placeholder: any; autocomplete: any; }[]} */
+  const inputsArray = [];
+  const childrenArray = Array.prototype.slice.call(formElement.childNodes);
+
+  for (const element of childrenArray) {
+    if (element.nodeName === 'INPUT' || element.nodeName === 'SELECT'
+    || element.nodeName === 'TEXTAREA') {
+      /** @type { { id: any; nodeName: any; name: any; placeholder: any; autocomplete: any;} } */
```

Comment:
if you skip the declaration and just jump straight to 

```js
inputsArray.push({
  id: ...
});
```

then you'll get typechecking  to match the array for free without having to redeclare the type in a comment :)